sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileSystems.java

```diff
@@ -470,10 +470,7 @@ static FileSystem getFileSystemInternal(String scheme) {
     String lowerCaseScheme = scheme.toLowerCase();
     Map<String, FileSystem> schemeToFileSystem = SCHEME_TO_FILESYSTEM.get();
     FileSystem rval = schemeToFileSystem.get(lowerCaseScheme);
-    if (rval != null) {
-      return rval;
-    }
-    return schemeToFileSystem.get(DEFAULT_SCHEME);
+    return checkNotNull(rval);
```

Comment:
I wouldn't use a `checkNotNull` but an explicit `IllegalArgumentException` saying no handler found for scheme `foo`.

Reply:
Will Fix this.